worker/src/main/scala/com/lucidchart/piezo/Worker.scala

```diff
@@ -118,6 +124,7 @@ object Worker {
       override def run() {
         logger.info("received shutdown signal")
         runSemaphore.release()
+        shutdownSemaphore.acquire()
```

Comment:
I'm not sure I understand the concurrency here. What is the purpose of shutdownSemaphore being acquired here? Is it to prevent the handler from exiting until main exits? Also, does acquiring this semaphore without releasing it block the program exiting?